cmd/frontend/middleware_body.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-HCP/pkg/api/arm"
+)
+
+func MiddlewareBody(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	switch r.Method {
+	case http.MethodPatch, http.MethodPost, http.MethodPut:
+		body, err := io.ReadAll(http.MaxBytesReader(w, r.Body, 1048576))
```

Comment:
can we state where the `1048576` value is coming from?  I believe ARM dictates this, but would be good to leave a breadcrumb or something.

Reply:
I actually don't know where that value comes from.  I copied it from ARO-RP, but [ARM states the max request body size is 4 MB](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#common-api-response-details), not 1 MB.  Jim Minter chose the 1 MB limit but never stated why, at least not in this repo.

I'll make it 4 MB and link to the ARM docs.